pkg/storegateway/bucket_test.go

```diff
@@ -480,260 +480,267 @@ func TestBucketIndexReader_ExpandedPostings(t *testing.T) {
 	tb := test.NewTB(t)
 	const series = 500
 
-	newTestBucketBlock := prepareTestBlock(tb, appendTestSeries(series))
-
-	t.Run("happy cases", func(t *testing.T) {
-		benchmarkExpandedPostings(test.NewTB(t), newTestBucketBlock, series)
-	})
-
-	t.Run("corrupted or undecodable postings cache doesn't fail", func(t *testing.T) {
-		b := newTestBucketBlock()
-		b.indexCache = corruptedPostingsCache{}
-
-		// cache provides undecodable values
-		matchers := []*labels.Matcher{labels.MustNewMatcher(labels.MatchRegexp, "i", "^.+$")}
-		refs, err := b.indexReader().ExpandedPostings(context.Background(), matchers, newSafeQueryStats())
-		require.NoError(t, err)
-		require.Equal(t, series, len(refs))
-	})
+	bucketBlockFactories := map[string]func() *bucketBlock{
+		"binary reader": prepareTestBlockWithBinaryReader(tb, appendTestSeries(series)),
+		"stream reader": prepareTestBlockWithStreamReader(tb, appendTestSeries(series)),
```

Comment:
This prepares two blocks even if just one of them will be ran. Maybe we can `prepareTestBlock` justonce, and instantiate two different readers here?

Reply:
That's right, but I'm not sure there's much value in spending further time on it. The non-streaming reader will be soon dropped.

Reply:
@colega has done a 👍 so assume he's good with these changes. Going to merge, but Oleg if you have any comment I'm sure Charles will follow up shortly ;)